exporter/exporterqueue/sized_queue.go

```diff
@@ -104,17 +114,20 @@ func (sq *sizedQueue[T]) pop() (context.Context, T, bool) {
 
 	for {
 		if sq.size > 0 {
-			ctx, el, elSize := sq.items.pop()
+			elCtx, el, elSize := sq.items.pop()
 			sq.size -= elSize
-			return ctx, el, true
+			sq.hasMoreSpace.Signal()
+			return elCtx, el, true
 		}
 
 		if sq.stopped {
 			var el T
 			return context.Background(), el, false
 		}
 
-		sq.hasElements.Wait()
+		// TODO: Change the Queue interface to return an error to allow distinguish between shutdown and context canceled.
+		// Ok to ignore the error, since the context.Background() will never be done.
+		_ = sq.hasMoreElements.Wait(context.Background())
```

Comment:
Do we need to switch `hasMoreElements` to the custom cond to resolve the TODO later?

Reply:
We need only to have "Read" return an error instead of bool, and have options like "ErrQueueShutdown" and "context.ErrDeadlineExceed" .

Reply:
If your suggestion was to still use the `sync.Cond` and change that when also solving the TODO, I think it is better. I did that.